python/mp/effects/effect.py

```diff
@@ -22,13 +22,28 @@ class Effect(abc.ABC):
     dm = DispatcherMapper()
 
     def __init__(self, *args, **kwargs):
+        # id will be assigned when the effect is attached to the mainloop
+        self.id = None
 
```

Comment:
I moved this here b/c some debugging print()s were failing b/c self.id wasn't set